src/test/java/specs/GherkinExampleSpecs.java

```diff
@@ -54,6 +62,38 @@
         });
       });
 
+      Supplier<List<String>> list = let(ArrayList::new);
+      scenario("behaviour of let in gherkin", () -> {
+        given("the data is set", () -> {
+          list.get().add("hello");
+        });
+
+        then("the data is still there", () -> {
+          assertThat(list.get().get(0), is("hello"));
```

Comment:
Before the fix, this was failing as the `let` was being freshened for both `given` and `then`